framework/source/class/qx/event/handler/DragDrop.js

```diff
@@ -768,21 +768,25 @@ qx.Class.define("qx.event.handler.DragDrop",
 
             // new drop target detected
             if (droppable && droppable != self.__dropTarget) {
-              // fire dragleave for previous drop target
-              if (self.__dropTarget) {
-                return self.__fireEvent("dragleave", self.__dropTarget, self.__dragTarget, false, e);
-              }
-
               self.__validDrop = true; // initial value should be true
               self.__dropTarget = droppable;
 
               var innerTracker = {};
               qx.event.Utils.catch(innerTracker, function () {
                 self.__validDrop = false;
               });
+              
+              // fire dragleave for previous drop target
+              if (self.__dropTarget) {
+                qx.event.Utils.then(innerTracker, function () {
+                  return self.__fireEvent("dragleave", self.__dropTarget, self.__dragTarget, false, e);
```

Comment:
This _looks_ like the `dragleave` is now issued on the wrong target, isn't it? At this point, `self.__dropTarget` has already been set to the new, droppable target, but then this code issues the `dragleave` on that new droppable target. Is it instead necessary to save `self.__dropTarget`'s value near line 771, and use the saved value here?

Reply:
yes, you're right again - 2secs, just going to double check the original before i push again